packages/bruno-electron/src/ipc/network/authorize-user-in-window.js

```diff
@@ -1,6 +1,10 @@
 const { BrowserWindow } = require('electron');
 const { preferencesUtil } = require('../../store/preferences');
 
+const matchesCallbackUrl = (url, callbackUrl) => {
+  return url ? new URL(url).host == new URL(callbackUrl).host : false;
+};
```

Comment:
Nope: my callback url is pointing to the same server that is auth url. It will pass this check and fail on missing 'code' even before the first redirect. 
![Screenshot from 2024-04-22 23-48-42](https://github.com/usebruno/bruno/assets/2108093/ed2a0e6e-46a7-4ff6-b5be-bb37335da164)

Reply:
Oh man, there are _lots_ of edge cases here 😅 

Comparing the `href` as you suggested earlier will not work as the href includes any query params in the URL, i.e. while it solves the trailing `/` problem it breaks for the below case:

```
callbackUrl: https://callback.url/endpoint
testUrl       : https://callback.url/endpoint?code=abcd

// testUrl.href = https://callback.url/endpoint?code=abcd
```

This is the reason I compared against hostname instead of href. I currently don't have a better idea to implement this, will think about it and come back later.

Ideas are welcome if I'm missing something 😇

Reply:
Wouldn't this suffice?
`testUrl.href.startswith(callbackUrl.href)`

The spec mandates the code param is _appended_ the the query component, so it should be at the end. Also luckily the fragment component (the url part after #) is not allowed, so it does not have to be handled.

Reply:
Thanks, that worked! I have pushed the change, would someone mind testing it out once?

I faced a weird problem testing locally, not sure if I introduced a bug or found an existing one 😅 

I'm able to generate auth tokens successfully now, but I can't make any requests - When I click the send request button, it just shows me the response from my OAuth server with the access token, it doesn't actually make a request using that token or display that response:
 
![Screenshot 2024-04-23 at 9 59 51 PM](https://github.com/usebruno/bruno/assets/36289388/bb9a7ab0-7c05-40d6-b670-f87ca4f2c61e)

Reply:
I wouldn't mind that. That's actually #1999

Reply:
Ah okay, thanks! 

So I guess that's it on this PR? Let me know if there's anything further to do!